aptos-move/framework/supra-framework/sources/vesting_without_staking.move

```diff
@@ -287,6 +287,98 @@ module supra_framework::vesting_without_staking {
         }
     }
 
+    public entry fun create_vesting_contract_with_amounts (
+        admin: &signer,
+        shareholders: vector<address>,
+        amounts: vector<u64>,
+        schedule_numerator: vector<u64>,
+        schedule_denominator: vector<u64>,
+        start_timestamp_secs: u64,
+        period_duration: u64,
+        withdrawal_address: address,
+        contract_creation_seed: vector<u8>,
+    ) acquires AdminStore {
+        assert!(!system_addresses::is_reserved_address(withdrawal_address),
+            error::invalid_argument(EINVALID_WITHDRAWAL_ADDRESS),);
+        assert_account_is_registered_for_apt(withdrawal_address);
+        assert!(vector::length(&shareholders) > 0,
+            error::invalid_argument(ENO_SHAREHOLDERS));
+        assert!(
+            vector::length(&shareholders) == vector::length(&amounts),
+            error::invalid_argument(ESHARES_LENGTH_MISMATCH),
+        );
+
+        // If this is the first time this admin account has created a vesting contract, initialize the admin store.
+        let admin_address = signer::address_of(admin);
+        if (!exists<AdminStore>(admin_address)) {
+            move_to(admin,
+                AdminStore {
+                    vesting_contracts: vector::empty<address>(),
+                    nonce: 0,
+                    create_events: new_event_handle<CreateVestingContractEvent>(admin),
+                });
+        };
+
+        // Initialize the vesting contract in a new resource account. This allows the same admin to create multiple
+        // pools.
+        let (contract_signer, contract_signer_cap) = create_vesting_contract_account(admin,
+            contract_creation_seed);
+        let contract_signer_address = signer::address_of(&contract_signer);
+        let i = 0;
+        let schedule = vector::empty<FixedPoint32>();
+        while (i < vector::length(&schedule_denominator)) {
+            let numerator = vector::borrow(&schedule_numerator, i);
+            let denominator = vector::borrow(&schedule_denominator, i);
+            let event = fixed_point32::create_from_rational(*numerator, *denominator);
+            vector::push_back(&mut schedule, event);
+            i = i + 1;
+        };
+
+        let vesting_schedule = create_vesting_schedule(schedule, start_timestamp_secs, period_duration);
+        let shareholders_map = simple_map::create<address, VestingRecord>();
+        let grant_amount = 0;
+        vector::for_each_ref(&amounts, |amount| {
+            coin::transfer<SupraCoin>(admin, contract_signer_address, *amount);
```

Comment:
This can be taken out of the loop, just use a single `transfer` for the entire grant amount, after checking that total grant is non-zero.